src/features/commentsSlice.ts

```diff
@@ -0,0 +1,66 @@
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import { getPostComments } from '../api/comments';
+import { Comment } from '../types/Comment';
+
+export type CommentState = {
+  comments: Comment[];
+  loading: boolean;
+  error: boolean;
+};
+
+const initialState: CommentState = {
+  comments: [],
+  loading: false,
+  error: false,
+};
+
+export const fetchComments = createAsyncThunk(
+  'comments/fetchComments',
+  (postId: number) => getPostComments(postId),
+);
+
+const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {
+    addNewComment: (state, action: PayloadAction<Comment>) => {
+      state.comments.push(action.payload);
+    },
+    removeComment: (state, action: PayloadAction<number>) => {
+      // eslint-disable-next-line no-param-reassign
+      state.comments = state.comments.filter(
+        comment => comment.id !== action.payload,
+      );
+    },
+    changeError: state => {
+      // eslint-disable-next-line no-param-reassign
+      state.error = true;
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(fetchComments.pending, state => {
+      return { ...state, loading: false };
```

Comment:
There is an error in the logic for the pending state. Typically, when a fetch request is pending, the loading state should be set to true, not false. This ensures that the UI can display a loading indicator while the request is in progress.